docs/topics/native/native-c-interop-stability.md

```diff
@@ -0,0 +1,180 @@
+native APIs to Kotlin.
+Just like above, the JetBrains team makes reasonable efforts to avoid breaking changes in such cases as well.
+
+## Importing third-party native libraries when the Xcode version doesn't match
+
+Apart from system libraries, Kotlin/Native allows importing third-party native libraries.
+For example,
+by using [CocoaPods integration](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-cocoapods-dsl-reference.html)
+or defining a [cinterops configuration](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-dsl-reference.html#cinterops).
+This has implications for the compatibility of your project with different Xcode versions.
+
+Every Kotlin version is compatible the most with a single Xcode version.
+You can find the versions
+[in this table](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-compatibility-guide.html#version-compatibility).
+This is the recommended version, which is tested the most with the corresponding Kotlin version. 
+Using other Xcode versions is often possible but might lead to problems, usually affecting importing third-party native
+libraries.
+
+### Using the Xcode version newer than recommended
+When using the Xcode version newer than recommended above, some Kotlin features might be broken.
+Importing third-party native libraries is affected by this the most.
+It often doesn't work at all with an unsupported version of Xcode.
+
+### Using the Xcode version older than recommended
+
+Typically, Kotlin works fine with older Xcode versions.
+There might be occasional issues, most frequently one of the two:
+- The Kotlin API of the native library refers to a non-existent type
+  (example: [KT-71694](https://youtrack.jetbrains.com/issue/KT-71694)).
+- The Kotlin API of the native library includes a type that belongs to a system library.
+  I.e., everything compiles well, but some system native type is added to your native library package.
+
+If you develop a Kotlin library, and it compiles with an older Xcode version fine, it is safe to publish (unless you
+[use types from the third-party native library in your Kotlin library API](#using-native-types-in-library-api)).
+
+## Using a transitive third-party native dependency
+
+When a Kotlin library used in your project imports a third-party native library as its implementation detail,
+your project also gets access to that native library.
+This happens because Kotlin/Native doesn't support the separation between `api` and `implementation` dependencies,
+so native libraries always end up being `api` dependencies.
+
+Using such a transitive native dependency is prone to more compatibility issues.
+For example, a change made by the Kotlin library developer might alter the Kotlin representation of the native library
+incompatibly, leading to compatibility issues when you update the Kotlin library.
```

Comment:
> leading to compatibility issues when you update the Kotlin library

I would expect to see the same issues as [here](https://github.com/JetBrains/kotlin-web-site/pull/4866/files#diff-bf4cddfca77f4bd73256d4db8da1649e9dad6043dd881431767a5e6dc388be5eR40-R43), is that correct? So you would either see your sources failing to compile, or your consumers might see the partial linkage behavior if you are a library author. Could this manifest as some other incompatibility?